test/conversion_test.go

```diff
@@ -947,3 +1136,162 @@ func TestPipelineRunCRDConversion(t *testing.T) {
 		t.Errorf("-want, +got: %v", d)
 	}
 }
+
+// TestBundleConversion tests v1beta1 bundle syntax converted into v1 since it has been
+// deprecated in v1 and it would be converted into bundle resolver in pipelineRef and
+// taskRef.
+func TestBundleConversion(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	t.Parallel()
+
+	c, namespace := setup(ctx, t, withRegistry, requireBundleFeatureFlags)
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
+
+	repo := fmt.Sprintf("%s:5000/tektonbundlessimple", getRegistryServiceIP(ctx, t, c, namespace))
+	ref, err := name.ParseReference(repo)
+	if err != nil {
+		t.Fatalf("Failed to parse %s as an OCI reference: %s", repo, err)
+	}
+
+	taskName := helpers.ObjectNameForTest(t)
+	pipelineName := helpers.ObjectNameForTest(t)
+	v1beta1TaskRunName := helpers.ObjectNameForTest(t)
+	task := parse.MustParseV1beta1Task(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  steps:
+  - name: hello
+    image: alpine
+    script: 'echo Hello'
+`, taskName, namespace))
+	pipeline := parse.MustParseV1beta1Pipeline(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  tasks:
+  - name: hello-world
+    taskRef:
+      resolver: bundles
+      params:
+      - name: bundle
+        value: %s
+      - name: name
+        value: %s
+`, pipelineName, namespace, repo, taskName))
+	v1beta1TaskRun := parse.MustParseV1beta1TaskRun(t, fmt.Sprintf(v1beta1TaskRunWithBundleYaml, v1beta1TaskRunName, namespace, taskName, repo))
+	v1TaskRunExpected := parse.MustParseV1TaskRun(t, fmt.Sprintf(v1TaskRunWithBundleExpectedYaml, v1beta1TaskRunName, namespace, repo, taskName, v1beta1TaskRunName))
+	v1beta1TaskRunRoundTripExpected := parse.MustParseV1beta1TaskRun(t, fmt.Sprintf(v1beta1TaskRunWithBundleRoundTripYaml, v1beta1TaskRunName, namespace, repo, taskName, v1beta1TaskRunName))
+
+	// Write the task and pipeline into an image to the registry in the proper format.
+	rawTask, err := yaml.Marshal(task)
+	if err != nil {
+		t.Fatalf("Failed to marshal task to yaml: %s", err)
+	}
+
+	rawPipeline, err := yaml.Marshal(pipeline)
+	if err != nil {
+		t.Fatalf("Failed to marshal task to yaml: %s", err)
+	}
+	img := empty.Image
+	taskLayer, err := tarball.LayerFromReader(bytes.NewBuffer(rawTask))
```

Comment:
I saw your comment about extracting this helper logic into a separate function since it's also used in https://github.com/tektoncd/pipeline/blob/main/test/tektonbundles_test.go, but that's a simple enough TODO that it's reasonable to address as part of this PR.

Reply:
Thanks Lee, updated. That does make it look nicer.